### PR TITLE
Nfdiv-2261 fix withdrawn notification for app2

### PR DESCRIPTION
**Pull request by lara-mpt**
### Change description ###

Prevent sending of withdrawn notification to app2 if app2 email not provided

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2261

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotification.java [hunk 1]
```diff
@@ -44,23 +45,25 @@ public void sendToApplicant1(final CaseData caseData, final Long id) {
 
     @Override
     public void sendToApplicant2(final CaseData caseData, final Long id) {
-        log.info("Sending application withdrawn notification to applicant 2 for: {}", id);
-        final Map<String, String> templateVars =
-            commonContent.mainTemplateVars(caseData, id, caseData.getApplicant2(), caseData.getApplicant1());
+        if (nonNull(caseData.getApplicant2().getEmail())) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotification.java:48]: I think we should check for both empty and null `ObjectUtils.isNotEmpty(email)`

File: src/test/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotificationTest.java [hunk 2]
```diff
@@ -209,4 +211,14 @@ void shouldSendEmailToJointApplicant2WithDissolutionContent() {
         );
         verify(commonContent).mainTemplateVars(data, 1234567890123456L, data.getApplicant2(), data.getApplicant1());
     }
+
+    @Test
+    void shouldNotEmailApplicant2IfEmailIsNotProvided() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotificationTest.java:216]: also worth adding test when email is empty

File: src/main/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotification.java [hunk 3]
```diff
@@ -44,23 +46,25 @@ public void sendToApplicant1(final CaseData caseData, final Long id) {
 
     @Override
     public void sendToApplicant2(final CaseData caseData, final Long id) {
-        log.info("Sending application withdrawn notification to applicant 2 for: {}", id);
-        final Map<String, String> templateVars =
-            commonContent.mainTemplateVars(caseData, id, caseData.getApplicant2(), caseData.getApplicant1());
+        if (nonNull(caseData.getApplicant2().getEmail()) && ObjectUtils.isNotEmpty(caseData.getApplicant2().getEmail())) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/notification/ApplicationWithdrawnNotification.java:49]: isNotEmpty checks for both null and empty so we can remove nonNull check

**Review by nitinprabhu — APPROVED** [review]
